executor/workflows/tasks.py

```diff
@@ -254,16 +259,66 @@ def test_workflow_notification(account_id, workflow, message_type):
             for step, all_task_results in all_step_executions.items():
                 for result in all_task_results:
                     playbook_execution_log = PlaybookExecutionLog(
-                        playbook=playbook,
-                        playbook_step=step,
-                        playbook_task_definition=result['task'].proto,
-                        playbook_task_result=result['task_result_proto'],
+                        playbook=p_proto,
+                        step=step.proto,
+                        task=result['task'].proto,
+                        task_execution_result=result['task_result_proto'],
                     )
                     pe_logs.append(playbook_execution_log)
 
         except Exception as exc:
             logger.error(f"Error occurred while running playbook: {exc}")
 
-        execution_output: [InterpretationProto] = playbook_execution_result_interpret(InterpreterType.BASIC_I, p_proto,
-                                                                                      pe_logs)
+        execution_output: [InterpretationProto] = playbook_execution_result_interpret(InterpreterType.BASIC_I, p_proto, pe_logs)
+        action_executor(account, workflow.actions[0], execution_output)
```

Comment:
In all the places where we are accessing a list, add an empty list check to protect from unwanted exceptions breaking the code.

Reply:
i have a lot of these array validations in the view before calling this function